### PR TITLE
replace into-stream with Readable.from

### LINKED ISSUE #288: Unicode emoji converted to broken chars in certain extreme long string responses
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.26.2

### Plugin version

7.0.2

### Node.js version

18.x

### Operating system

Windows

### Description

Download a 200kb yaml plain text file with `fastify/compress`. Certain lines of the yaml contains illegal chars. 

image

As you can see, on the left side (without compress), icons in the small green box are same. 
But on the right side, one of the icon becomes 2 strange question marks.

The content encoding is `br` (I tried with `gzip`, same error)

image

Any thoughts, what can be the root cause? 

### Steps to Reproduce

```js
const fastify = new Fastify({})
fastify.register(fastifyCompress)

fastify.get('/subscription', async (req, reply) => {
  const subscription = await handler.downloadClashSubscriptionAsync(config)

  // await require('fs').promises
  //   .writeFile('/out.txt', subscription, 'utf-8')  // <--- the file content is okay

  return reply.send(subscription)  // <--- the file content is corrupt
})
```

### Expected Behavior

Dont expect the two question marks appear in the content.

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #288

### CODE REVIEW

**Review by climba03003 — APPROVED** [review]

File: test/issue-288.test.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const fastifyCompress = require('..')
+const fsPromises = require('fs').promises
+const { join } = require('path')
+const { fetch } = require('undici')
+
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:9]:
```suggestion

const longStringWithEmoji = new Array(5_000)
  .fill("0")
  .map(() => {
    const part1 = new Array(Math.floor(Math.random() * 10) + 2)
      .fill("A")
      .join("🍃");
    return part1 + "- FASTIFY COMPRESS,🍃 FASTIFY COMPRESS";
  })
  .join("\n");
```

File: test/issue-288.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const fastifyCompress = require('..')
+const fsPromises = require('fs').promises
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:6]:
```suggestion
```

File: test/issue-288.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const { fetch } = require('undici')
+
+test('should not corrupt the file content', async (t) => {
+  const fastify = new Fastify()
+  t.teardown(() => fastify.close())
+
+  fastify.register(async (instance, opts) => {
+    await fastify.register(fastifyCompress)
+    instance.get('/issue', async (req, reply) => {
+      const longStringWithEmoji = await fsPromises.readFile(join(__dirname, './test.txt'), 'utf-8')
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:17]:
```suggestion
```

File: test/issue-288.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+      return longStringWithEmoji // <--- the file content is corrupt
+      // search for "hydra.alibaba.com" will see 2 wired question marks instead of emoji
+    })
+  })
+
+  fastify.get('/good', async (req, reply) => {
+    const longStringWithEmoji = await fsPromises.readFile(join(__dirname, './test.txt'), 'utf-8')
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:25]:
```suggestion
    const longStringWithEmoji = await fsPromises.readFile(join(__dirname, './test.txt'), 'utf-8')
```

File: test/issue-288.test.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  const fastify = new Fastify()
+  t.teardown(() => fastify.close())
+
+  fastify.register(async (instance, opts) => {
+    await fastify.register(fastifyCompress)
+    instance.get('/issue', async (req, reply) => {
+      const longStringWithEmoji = await fsPromises.readFile(join(__dirname, './test.txt'), 'utf-8')
+
+      return longStringWithEmoji // <--- the file content is corrupt
+      // search for "hydra.alibaba.com" will see 2 wired question marks instead of emoji
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:20]:
```suggestion
      // Expect response should not contain any �
```

File: test/issue-288.test.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+test('should not corrupt the file content', async (t) => {
+  const fastify = new Fastify()
+  t.teardown(() => fastify.close())
+
+  fastify.register(async (instance, opts) => {
+    await fastify.register(fastifyCompress)
+    instance.get('/issue', async (req, reply) => {
+      const longStringWithEmoji = await fsPromises.readFile(join(__dirname, './test.txt'), 'utf-8')
+
+      return longStringWithEmoji // <--- the file content is corrupt
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:19]:
```suggestion
      return longStringWithEmoji
```

File: test/issue-288.test.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      // search for "hydra.alibaba.com" will see 2 wired question marks instead of emoji
+    })
+  })
+
+  fastify.get('/good', async (req, reply) => {
+    const longStringWithEmoji = await fsPromises.readFile(join(__dirname, './test.txt'), 'utf-8')
+
+    return longStringWithEmoji // <--- the file content is ok
+    // search for "hydra.alibaba.com" will see emoji
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:28]:
```suggestion
    return longStringWithEmoji
    // Expect response should not contain any �
```

**Comment by climba03003** [comment]
`Readable.from` doesn't like a directly replacement.
`into-stream` support more data types than expected.

File: test/issue-288.test.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  await fastify.listen({ port: 0 })
+
+  const { port } = fastify.server.address()
+  const url = `http://localhost:${port}`
+  const response = await fetch(`${url}/issue`)
+  const response2 = await fetch(`${url}/good`)
+  const body = await response.text()
+  const body2 = await response2.text()
+  t.equal(body, body2)
```
Comment (stanleyxu2005) [inline test/issue-288.test.js:39]:
```suggestion
  t.equal(body, body2)
  t.equal(body.indexOf('�'), -1)
  t.equal(body2.indexOf('�'), -1)
```

**Comment by stanleyxu2005** [comment]
@mcollina I suggest to generate a long string instead of loading from `test.txt`

File: test/issue-288.test.js (hunk 8 above)
Comment (stanleyxu2005) [reply test/issue-288.test.js:39]: Ensure number of emoji is valid

File: package.json [hunk 9]
```diff
@@ -26,7 +25,8 @@
     "standard": "^17.1.0",
     "tap": "^16.3.7",
     "tsd": "^0.30.0",
-    "typescript": "^5.1.6"
+    "typescript": "^5.1.6",
+    "undici": "^5.28.3"
```
Comment (Eomm) [inline package.json:29]: Should it be devDependancy?

**Pull request closed without merging**